tests/db/test_voter.py

```diff
@@ -297,6 +297,120 @@ def test_updating_block_number_must_be_atomic(self):
         pass
 
 
+class TestBlockElection(object):
+
+    def test_quorum(self, b):
+        # create a new block
+        test_block = b.create_block([])
+
+        # simulate a federation with four voters
+        test_block['block']['voters'] = ['a', 'b', 'c', 'd']
+
+        # fake "yes" votes
+        valid_vote = b.vote(test_block, 'abc', True)
+
+        # fake "no" votes
+        invalid_vote = b.vote(test_block, 'abc', False)
+
+        # fake "yes" votes with incorrect signatures
+        improperly_signed_valid_vote = b.vote(test_block, 'abc', True)
+        improperly_signed_valid_vote['vote']['lol'] = 'this should ruin things'
+
+        # test unanimously valid block
+        test_block['block']['votes'] = [valid_vote, valid_vote, valid_vote, valid_vote]
+        assert b.block_election_status(test_block) == 'valid'
+
+        # test partial quorum situations
+        test_block['block']['votes'] = [valid_vote, valid_vote]
+        assert b.block_election_status(test_block) == 'undecided'
+        #
+        test_block['block']['votes'] = [valid_vote, valid_vote, valid_vote]
+        assert b.block_election_status(test_block) == 'valid'
+        #
+        test_block['block']['votes'] = [invalid_vote, invalid_vote]
+        assert b.block_election_status(test_block) == 'invalid'
+
+        # test unanimously valid block with one improperly signed vote -- should still succeed
+        test_block['block']['votes'] = [valid_vote, valid_vote, valid_vote, improperly_signed_valid_vote]
+        assert b.block_election_status(test_block) == 'valid'
+
+        # test unanimously valid block with two improperly signed votes -- should fail to have quorum
+        test_block['block']['votes'] = [valid_vote, valid_vote,
```

Comment:
This is a situation in which quorum will never be reached and we cannot have that situation.

To be on the safe side we should define that vote is invalid if:
- the vote is correct and it voted the block invalid
- the vote is not correct (invalid signature, ...)

Better to just requeue the transactions and try again then having the block forever stuck in an undecided state